src/building/java/net/remmintan/mods/minefortress/blocks/building/FortressBuildingBlockEntity.kt

```diff
@@ -129,6 +129,12 @@ class FortressBuildingBlockEntity(pos: BlockPos?, state: BlockState?) :
     override fun getFurnace(): FurnaceBlockEntity? =
         furnaceBlockPos?.let { this.getWorld()?.let { w -> w.getBlockEntity(it) as? FurnaceBlockEntity } }
 
+    override fun getFurnacePos(): BlockPos? =
+        this.blockData?.referenceState
+            ?.filter { it.blockState.block?.translationKey == "block.minecraft.furnace" }
+            ?.map { it.pos }
+            ?.get(0)
```

Comment:
with this soltuion you could extend your implementation to support several Furnaces per building. Would be awesome to see that

Reply:
I had actually played around with multiple furnaces before making this PR, but I figured I would keep it the way it was originally. I'm glad to hear you're open to multiple furnaces per building, I think it would be neat too! 

I added the ability to use all furnaces from all blacksmiths buildings, but the number of furnaces usable is limited to the number of blacksmith units that have been hired.